src/backend/base/langflow/api/v1/endpoints.py

```diff
@@ -371,6 +382,170 @@ async def on_disconnect() -> None:
+                    status_code=400,
+                    detail=f"Invalid file name format: {file.filename}. Must be 'ComponentName::input_name::filename'",
+                )
+
+            component_name = match.group(1)
+            input_name = match.group(2)
+            original_filename = match.group(3)
+
+            # Save file and get path
+            file_name = f"{uuid.uuid4()}_{original_filename}"
+            await storage_service.save_file(flow_id=str(flow.id), file_name=file_name, data=await file.read())
+
+            # Add file path to tweaks
+            file_path = storage_service.build_full_path(flow_id=str(flow.id), file_name=file_name)
+
+            # Initialize component tweaks if not exists
+            if component_name not in file_tweaks:
+                file_tweaks[component_name] = {input_name: {}}
+
+            file_tweaks[component_name][input_name]["file_path"] = file_path
+
+        # Merge file tweaks with existing tweaks
+        if input_request.tweaks is None:
+            input_request.tweaks = Tweaks()
+        input_request.tweaks.update(file_tweaks)
+
+    if stream:
+        asyncio_queue: asyncio.Queue = asyncio.Queue()
+        asyncio_queue_client_consumed: asyncio.Queue = asyncio.Queue()
+        event_manager = create_stream_tokens_event_manager(queue=asyncio_queue)
+        main_task = asyncio.create_task(
+            run_flow_generator(
+                flow=flow,
+                input_request=input_request,
+                api_key_user=api_key_user,
+                event_manager=event_manager,
+                client_consumed_queue=asyncio_queue_client_consumed,
+            )
+        )
+
+        async def on_disconnect() -> None:
+            logger.debug("Client disconnected, closing tasks")
+            main_task.cancel()
+
+        return StreamingResponse(
+            consume_and_yield(asyncio_queue, asyncio_queue_client_consumed),
+            background=on_disconnect,
+            media_type="text/event-stream",
+        )
+
+    try:
+        result = await simple_run_flow(
+            flow=flow,
+            input_request=input_request,
+            stream=stream,
+            api_key_user=api_key_user,
+        )
+        end_time = time.perf_counter()
+        background_tasks.add_task(
+            telemetry_service.log_package_run,
+            RunPayload(
+                run_is_webhook=False,
+                run_seconds=int(end_time - start_time),
+                run_success=True,
+                run_error_message="",
+            ),
+        )
+
+    except ValueError as exc:
+        background_tasks.add_task(
+            telemetry_service.log_package_run,
+            RunPayload(
+                run_is_webhook=False,
+                run_seconds=int(time.perf_counter() - start_time),
+                run_success=False,
+                run_error_message=str(exc),
+            ),
+        )
+        if "badly formed hexadecimal UUID string" in str(exc):
```

Comment:
nit: maybe we could check the flow id with something like `uuid_obj = uuid.UUID(uuid_str)` in a try block? basically just thinking about the scenario where upstream they were to change the format of the text error message in any way, if we could catch this error without parsing the text

Reply:
Done.